Automattic/wp-calypso
# AI Logo Generator: Add error handling

Closes #86479

## Proposed Changes

* Adds store entries for all errors
* Shows error messages where necessary

## Testing Instructions



To test the errors it is necessary to force them individually.
* Select a site with a paid plan
* First check that it works as normal
* Now for each of the following functions on the UseLogoGenerator hook, manually throw an error inside their `try` block, one by one:
* `generateFirstPrompt`: To trigger the first prompt, go to the localStorage and remove the history of the site. It should be an entry like `logo-history-<site id>`. Then open the modal. It should show a message about failing to load the account settings with a retry button.
*  `enhancePrompt`: To trigger this, just click the Enhance prompt button. A red string with an error message should be visible.
* `generateImage`: To trigger this, click on the Generate button. A red string with an error message should be visible.
* `saveLogo`: Just generate an image, it tries to save the logo automatically. A red string with an error is expected.
* `applyLogo`: Try to use the logo on the site. A red string with an error is expected.
* `fetchAiAssistantFeature`: This one is on the actions file. The same message as the first case about failing to load the account settings should be displayed.

## Pre-merge Checklist
